internal/types/path.go

```diff
@@ -17,17 +17,87 @@ package types
 import (
 	"fmt"
 	"strconv"
+	"strings"
 )
 
+// Path represents field path type. It should be used wherever we work with paths or dot notation.
+// Path should be stored and passed as value. All of its methods return new values, not modifying the receiver's state.
+type Path struct {
+	s []string
+}
+
+// NewPath returns Path from a strings slice.
+func NewPath(path []string) Path {
+	p := Path{s: make([]string, len(path))}
+	copy(p.s, path)
+	return p
+}
+
+// NewPathFromString returns Path from path string. Path string should contain fields separated with '.'.
+func NewPathFromString(s string) Path {
+	path := strings.Split(s, ".")
+
+	return NewPath(path)
+}
+
+// String returns dot-separated path value.
+func (p Path) String() string {
+	return strings.Join(p.s, ".")
+}
+
+// Len returns path length.
+func (p Path) Len() int {
+	return len(p.s)
+}
+
+// Slice returns path values array.
+func (p Path) Slice() []string {
+	path := make([]string, len(p.s))
+	copy(path, p.s)
+	return path
+}
+
+// Suffix returns the last path element.
+func (p Path) Suffix() string {
+	if len(p.s) == 0 {
+		return ""
+	}
+	return p.s[p.Len()-1]
```

Comment:
In some cases you use `len(p.s)` to get the length (e.g. three lines above). In others, like here, you use `p.(Len)`. Let's make it more consistent and use the same approach everywhere in this file?

Reply:
Fixed.